metricflow-semantics/metricflow_semantics/model/semantics/linkable_element.py

```diff
@@ -105,9 +121,20 @@ def path_key(self) -> ElementPathKey:  # noqa: D102
     def reference(self) -> DimensionReference:  # noqa: D102
         return DimensionReference(element_name=self.element_name)
 
+    @property
+    @override
+    def derived_from_semantic_models(self) -> Sequence[SemanticModelReference]:
+        semantic_model_references = set()
+        if self.semantic_model_origin:
+            semantic_model_references.add(self.semantic_model_origin)
+        for join_path_item in self.join_path:
+            semantic_model_references.add(join_path_item.semantic_model_reference)
+
+        return sorted(semantic_model_references, key=lambda reference: reference.semantic_model_name)
+
 
 @dataclass(frozen=True)
-class LinkableEntity:
+class LinkableEntity(LinkableElement, SemanticModelDerivation):
```

Comment:
You don't need both of these, right, just LinkableElement?

Reply:
Right - updated.